### ISSUE TITLE
Can't add unit tests to a project?

### ISSUE THREAD

**Issue by underbluewaters** [seasketch/geoprocessing#254]
I can't add arbitrary thing.test.ts files to functions/ and run these unit tests using `npm test`. The package.json file seems to be doing a lot of gymnastics to start up servers so it seems the tests need to be in an explicit grouping? It's not obvious what to do here for true unit tests for isolated functions that aren't part of a geoprocessing handler.

**Comment by twelch** [comment]
Good catch, yes you'll need to put test into one of the `unit`, `smoke`, or `e2e` test groups.  Like:

/**
 * @jest-environment node (optional if need full node environment such as for geoblaze tests)
 * @group unit
 */

**Comment by twelch** [comment]
It's not just about the gymnastics of running servers, but also taking control over exactly what tests get run locally vs when committing vs in CI, etc.
```
"test:unit": "npm run __test -- --group=unit --passWithNoTests"
```

**Comment by underbluewaters** [comment]
Would it make sense to have unlabelled tests go into a default grouping? I was able to add a group and get it to run in the end. It is nice to be able to specify what type of tests I want to run, but it adds another gotcha to work out if the author hasn't run into it before.

**Comment by avmey** [comment]
@twelch, definitely useful for testing the gp framework, is it also required in project space? I don't think there's any commit or github CI checks for project repos. For example, if I switch my `"test": "npm run test:unit && npm run test:smoke"` command in `fsm-nearshore-reports` to `"test": "start-server-and-test start-data 8080 'npm run __test -- --runInBand --silent'"` it catches and passes all tests (including ones ungrouped). Let me know if there's something I'm missing/ some tests in project space that shouldn't run there.

**Comment by twelch** [comment]
gp version 7, once released, uses Vitest which gets rid of test groups.  Both unit tests and smoke tests will run together under the one `test` command.  And if you want to run just one or the other you can use test name matching and for example put 'Smoke' in your test names.  But in practice, unless I'm developing new analysis functions in project-space I don't usually have unit tests for a project, just smoke tests.

This should resolve this issue
